tpm/tss2/tss2_test.go

```diff
@@ -0,0 +1,359 @@
+package tss2
+
+import (
+	"encoding/asn1"
+	"encoding/pem"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// key generated using:
+//
+//	openssl genpkey -provider tpm2 -algorithm RSA -out rsa-key.pem
+var rsaTSS2PEM = `-----BEGIN TSS2 PRIVATE KEY-----
+MIICEgYGZ4EFCgEDoAMBAQECBEAAAAEEggEYARYAAQALAAYAcgAAABAAEAgAAAAA
+AAEAxaLvqN+HkUsj83CQxxKwjB/OODnMrLzCly3917Fv0iSehV0sjkrW+9kpksYL
+5nAdhT/EhfOGupmq8hpEghEFUvocarGf/CtT7ra/FE1d6JT8F6bPe9lIUNcUhObx
+Msw8JU3f/uPNurlA/yXb8ZTlGSHoXUrYbiDF36EJUKm2nc/QkdB4SyXrAaUjTFAC
+8ifqb3cvtx1KrLGhbP+dQ1A2ytK8FaMauhOGp3gIlAOOVwcRg/DvaxzVBlKRW75u
+21r6/w5Wo71IdeEPB/qlCfY3CGvk3in6Wf27gjS+vVa/gx9N9sV7Sq/QX5rWnyol
+x5cb6miCxLAWrchU80bGi0h+BwSB4ADeACCMbeh27b01dSmGrtftJnU9ZJaXr6tr
+us/XbpEh+QY/NQAQZlRBv8rA2Gicwnf5gbZcBGVAOxp/EWZH+MVEKQOk4EVM+ACa
+EQRWNdlr/A/9lPyOZIWw4tajkfKZgfr0ia0kAK4Cfb3cTnXnsAp59/kLG35JM2gQ
+/Xvm/CAxtvmVNPF6OrmWhNhoOHbveEok2nVKyr0WM5WQSCnnZH4DpyHtA3bXa2rb
+aPU0KW6T7K0wYsGI0FHC/3arTjlVmXC+mze86n32TkO0rhQiAyn8jS6tbUe6b3jW
+x7YfKP6g
+-----END TSS2 PRIVATE KEY-----`
+
+// key generated using:
+//
+//	openssl ecparam -provider tpm2 -name prime256v1 -genkey -noout -out ec-key.pem
+var p256TSS2PEM = `-----BEGIN TSS2 PRIVATE KEY-----
+MIHwBgZngQUKAQOgAwEBAQIEQAAAAQRYAFYAIwALAAYAcgAAABAAEAADABAAIO7L
+aur7h2XAyrZTA8g6QMksNNoUvkMZ4xnjVUSn3k+bACDHuNRZDInoD5Nts7WUos0k
+Oe0/tF/HfhfSQTqsHo/rKgSBgAB+ACBr9xn6R2V13ErShb75o+EyMqtFsTysp24f
+VNZ7IWfEBAAQebW0tKoqBmNr/ZGTt2jAEO/gdLEuZ+TkiWYf3h8Jcc0bUrj6lA9I
+W6fVV4B/ZtnADx9/YGB9FBY8Bu07W7m+PorVTCbXFfOAFmSUg3eB0bgb2TRtFevZ
+izcX
+-----END TSS2 PRIVATE KEY-----`
+
+// key generated using:
+//
+//	tpm2_createprimary -c primary.ctx
+//	tpm2_create -C primary.ctx -G ecc -u obj.pub -r obj.priv
+//	tpm2_encodeobject -C primary.ctx -u obj.pub -r obj.priv -o obj.pem
+var p256EmptyAuthFalse = `-----BEGIN TSS2 PRIVATE KEY-----
+MIHwBgZngQUKAQOgAwEBAAIEQAAAAQRYAFYAIwALAAYAcgAAABAAEAADABAAIIgq
+1VllQRCT45GbLlp1Wud0jiSfojBwp1MYljWMw1T7ACAblgTFkwvSMnzpArA8GjVP
+ULHy7pJubvS2W7TxmzclRQSBgAB+ACDVXoK8RpE5XxjBcfeHpip9Dz2j7AUj0oE1
+RKlDg/+dYgAQd9c9mgioJc8wFL1zaU4viH1fq3fObbfZF/L8oLrLv6u3Pg8qeGzf
+ePVypgEUeJGw68er7UZb4ZSVfoGId6KLX9JE7IwyBkRWLhBU3sLANdgjTqlXUhAD
+mnYo
+-----END TSS2 PRIVATE KEY-----`
+
+func TestParsePrivateKey(t *testing.T) {
+	parsePEM := func(s string) []byte {
+		block, _ := pem.Decode([]byte(s))
+		return block.Bytes
+	}
+
+	type args struct {
+		derBytes []byte
+	}
+	tests := []struct {
+		name      string
+		args      args
+		want      *TPMKey
+		assertion assert.ErrorAssertionFunc
+	}{
+		{"ok rsa", args{parsePEM(rsaTSS2PEM)}, &TPMKey{
+			Type:      oidLoadableKey,
+			EmptyAuth: true,
+			Parent:    1073741825,
```

Comment:
To sort of self-document this, this could become a hex const for `0x40000001`, e.g. `tpm2.HandleOwner`, which is the expected (default) value for a key created with the primary as the parent.